modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaScreenContributorClassCheck.java

```diff
@@ -0,0 +1,168 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.source.formatter.parser.JavaClass;
+import com.liferay.source.formatter.parser.JavaClassParser;
+import com.liferay.source.formatter.parser.JavaMethod;
+import com.liferay.source.formatter.parser.JavaTerm;
+import com.liferay.source.formatter.parser.JavaVariable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Tamyris Bernardo
+ */
+public class UpgradeJavaScreenContributorClassCheck extends BaseUpgradeCheck {
+
+	@Override
+	protected String format(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		JavaClass javaClass = JavaClassParser.parseJavaClass(fileName, content);
+
+		String oldJavaClassContent = javaClass.getContent();
+
+		Matcher matcher = _screenContributorPattern.matcher(
+			oldJavaClassContent);
+
+		if (!matcher.find()) {
+			return content;
+		}
+
+		String javaTermContent;
+
+		List<String> oldVariables = new ArrayList<>();
+
+		List<String> oldMethods = new ArrayList<>();
+
+		for (JavaTerm childJavaTerm : javaClass.getChildJavaTerms()) {
+			if (childJavaTerm.isJavaVariable()) {
+				JavaVariable javaVariable = (JavaVariable)childJavaTerm;
+
+				javaTermContent = javaVariable.getContent();
+
+				oldVariables.add(javaTermContent);
+			}
+			else if (childJavaTerm.isJavaMethod()) {
+				JavaMethod javaMethod = (JavaMethod)childJavaTerm;
+
+				javaTermContent = javaMethod.getContent();
+
+				oldMethods.add(javaTermContent);
+			}
+		}
+
+		return StringUtil.replace(
+			content, oldJavaClassContent,
+			_formatContent(
+				matcher, oldJavaClassContent, oldMethods, oldVariables));
+	}
+
+	protected String[] getNewImports() {
+		return new String[] {
+			"com.liferay.configuration.admin.display.ConfigurationScreen",
+			"com.liferay.configuration.admin.display." +
+				"ConfigurationScreenWrapper",
+			"com.liferay.portal.settings.configuration.admin.display." +
+				"PortalSettingsConfigurationScreenFactory"
+		};
+	}
+
+	private static List<String> _formatMethodsLines(List<String> methods) {
+		List<String> methodsLines = new ArrayList<>();
+
+		for (String method : methods) {
+			String[] methodLines = StringUtil.splitLines(method);
+
+			for (int i = 0; i < methodLines.length; i++) {
+				methodLines[i] = StringUtil.removeFirst(
+					methodLines[i], StringPool.TAB);
+
+				if (StringUtil.equals(
+						methodLines[i], StringPool.CLOSE_CURLY_BRACE)) {
+
+					methodLines[i] = methodLines[i] + StringPool.NEW_LINE;
+				}
+
+				if (methodLines[i].isEmpty()) {
+					methodsLines.set(
+						methodsLines.size() - 1,
```

Comment:
@nicolas-sss, have you considered the scenario in which the `methodsLines` is empty? Won't it generate exceptions?

Reply:
@natocesarrego Yeah, I have, that will never happen because the return of what is a method always will be the start of the method. So at least one line will have already been added to the `methodsLines` list.

Reply:
We agreed on offline and It's approved now.